docs/source/developers/java/building.rst

```diff
@@ -0,0 +1,224 @@
+    $ export JAVA_HOME=<absolute path to your java home>
+    $ mkdir -p java-dist java-native-c
+    $ cd java-native-c
+    $ cmake \
+        -DCMAKE_BUILD_TYPE=Release \
+        -DCMAKE_INSTALL_LIBDIR=lib \
+        -DCMAKE_INSTALL_PREFIX=../java-dist \
+        ../java/c
+    $ cmake --build . --target install
+    $ ls -latr ../java-dist/lib
+    |__ libarrow_cdata_jni.dylib
+
+To build other JNI libraries:
+
+.. code-block::
+
+    $ cd arrow
+    $ brew bundle --file=cpp/Brewfile
+    Homebrew Bundle complete! 25 Brewfile dependencies now installed.
+    $ export JAVA_HOME=<absolute path to your java home>
+    $ mkdir -p java-dist java-native-cpp
+    $ cd java-native-cpp
+    $ cmake \
+        -DARROW_BOOST_USE_SHARED=OFF \
+        -DARROW_BROTLI_USE_SHARED=OFF \
+        -DARROW_BZ2_USE_SHARED=OFF \
+        -DARROW_GFLAGS_USE_SHARED=OFF \
+        -DARROW_GRPC_USE_SHARED=OFF \
+        -DARROW_LZ4_USE_SHARED=OFF \
+        -DARROW_OPENSSL_USE_SHARED=OFF \
+        -DARROW_PROTOBUF_USE_SHARED=OFF \
+        -DARROW_SNAPPY_USE_SHARED=OFF \
+        -DARROW_THRIFT_USE_SHARED=OFF \
+        -DARROW_UTF8PROC_USE_SHARED=OFF \
+        -DARROW_ZSTD_USE_SHARED=OFF \
+        -DARROW_JNI=ON \
+        -DARROW_PARQUET=ON \
+        -DARROW_FILESYSTEM=ON \
+        -DARROW_DATASET=ON \
+        -DARROW_GANDIVA_JAVA=ON \
+        -DARROW_GANDIVA_STATIC_LIBSTDCPP=ON \
+        -DARROW_GANDIVA=ON \
+        -DARROW_ORC=ON \
+        -DARROW_PLASMA_JAVA_CLIENT=ON \
+        -DARROW_PLASMA=ON \
+        -DCMAKE_BUILD_TYPE=Release \
+        -DCMAKE_INSTALL_LIBDIR=lib \
+        -DCMAKE_INSTALL_PREFIX=../java-dist \
+        -DCMAKE_UNITY_BUILD=ON \
+        -Dre2_SOURCE=BUNDLED \
+        -DBoost_SOURCE=BUNDLED \
+        -Dutf8proc_SOURCE=BUNDLED \
+        -DSnappy_SOURCE=BUNDLED \
+        -DORC_SOURCE=BUNDLED \
+        -DZLIB_SOURCE=BUNDLED \
+        ../cpp
+    $ cmake --build . --target install
+    $ ls -latr  ../java-dist/lib
+    |__ libarrow_dataset_jni.dylib
+    |__ libarrow_orc_jni.dylib
+    |__ libgandiva_jni.dylib
+
+Building Arrow JNI Modules
+--------------------------
+
+To compile the JNI bindings, use the ``arrow-c-data`` Maven profile:
+
+.. code-block::
+
+    $ cd arrow/java
+    $ mvn -Darrow.c.jni.dist.dir=../java-dist/lib -Parrow-c-data clean install
+
+To compile the JNI bindings for ORC / Gandiva / Dataset, use the ``arrow-jni`` Maven profile:
+
+.. code-block::
+
+    $ cd arrow/java
+    $ mvn -Darrow.cpp.build.dir=../java-dist/lib -Parrow-jni clean install
+
+IDE Configuration
+=================
+
+IntelliJ IDE
+------------
+
+Go to open java project and select java folder.
+
+This is the initial view for java project loaded with default profiles:
+
+.. image:: img/java_welcome.png
+   :alt: An example image to opening java project
+
+Let's create our maven configuration: java-dataset
+
+.. image:: img/java_mvn_configuration.png
+   :alt: An example image to create new maven configurations and parameters needed
```

Comment:
Specifically, we should say which settings need to be changed to which values.

Also: do these `alt` blocks show up in the text? I think we should use the [figure directive](https://docutils.sourceforge.io/docs/ref/rst/directives.html#figure) which allows a caption.

Reply:
Changed